src/update_agent/mod.rs

```diff
@@ -370,7 +372,7 @@ pub(crate) struct UpdateAgent {
     /// Update strategy.
     strategy: UpdateStrategy,
     /// Current status for agent state machine.
-    state: UpdateAgentState,
+    state: Arc<RwLock<UpdateAgentState>>,
```

Comment:
(Also added as a comment in the code) I believe we need to use an `Arc` because consumers of this field will likely need to own it (e.g. consumers in futures).